src/utils.ts

```diff
@@ -49,14 +49,22 @@ export function replaceDDLNamePlaceholder(ddl: string, placeholder: string, repl
 	return undefined;
 }
 
-export function validateName(name: string): string {
+export function validateName(name: string): string | undefined {
 	if (name && /^[a-z0-9]{4,253}$/.test(name)) {
 		return undefined;
 	} else {
 		return 'The entered name does not comply with the naming conventions. ([a-z0-9] and length of 4-253 characters)';
 	}
 }
 
+export function ensureValidEnvironmentVariableName(name: string): string | undefined {
+	if (name && /^[A-Z]{1}[A-Z0-9_]{3,252}$/.test(name)) {
```

Comment:
from where the rule is coming from? What are the odds that it evolves? is there an external library to check for it?

Reply:
The rules have been defined by @rareddy in the last dv tooling call. This regexp just tests for those rules. I think it is related to some Spring Boot rules.

Reply:
Yes, ENV properties are typically in all caps, see some general discussion [1] Moreover it is the convention that Spring Boot uses to that Teiid is built upon. Also, there are no dot(.) allowed in the ENV properties, and Spring Boot expects these properties need to be replaced with underscore ( _ )

[1] https://discuss.codecademy.com/t/is-it-just-a-convention-or-all-environment-variables-have-to-be-uppercase/386240

Reply:
these ones are explaining only CAPITAL letters. what about the 4-253 size limits?

Reply:
We defined it like this.

Reply:
why? it is related to OpenShift resource limit like for Virtual Database definition name?
We need to understand why and be able to justify it. Otherwise, a user will modify it outside of the tooling, not respecting it and if it is working he/she will expect that the tooling support it and will report a bug.

Reply:
is possible at least show up some hint example how required name should looks like? and I think it would be nice to have some small note about these all restrictions inside readme?

Reply:
The validation message shows the restrictions partially as regexp and also human readable length definitions. I agree that documentation should contain a description of these validation rules in the corresponding function descriptions.

Reply:
> why? it is related to OpenShift resource limit like for Virtual Database definition name?
> We need to understand why and be able to justify it. Otherwise, a user will modify it outside of the tooling, not respecting it and if it is working he/she will expect that the tooling support it and will report a bug.

@rareddy I will leave that to you.

Reply:
There are no real size limits defined by Linux for environment variables, but I think 256 is a sensible size alternative. I also think if there is a variable that is more than 256, that probably is generated by some machine not human, so then there is no association with this plugin so we should not be too concerned about but if it is already getting to a serious usability issues.

Reply:
you are mentioning 256 but in code it is 254.

The use case of opening a generated vdb with the VS Code extension is not a use case that we want to support? We will support only things that are created from this VS Code extension?

Reply:
Ok on 254. I still do not have a use case where the VDB is generated, Right now it is purely hand-coded.  Even when/if we have generated VDB, Teiid yet to define with a property name in the Teiid configuration that goes beyond 50+ chars, much less 254 characters. So, IMO this more than fine.

Reply:
I guess one could argue that syndesis does generate a VDB, but the property comment is still valid

Reply:
I think that whole discussion is about a rare edge case nobody will ever face. Who manually defines an environment variable name with 200+ characters length? Whenever the first user (not you, Aurelien :) ) files an issue we can start thinking about that.

Reply:
guys, please remember what we have discussed endless amount of time:
subsequent improvement shouldn't stop development; hopefully, it shouldn't even slow it down.

this is a corner case. add a comment, a jira, and add it even in the release notes if you want, but please, don't waste time of this.

Reply:
Ok. So lets get this PR merged then. @apupier feel free to raise a Jira on that matter so we can track it.

Reply:
the 256 vs 254 was just an example of why we are going in wrong directions.

There is the same problem for minimal size, given that there is no constraints outside of VS Code extension, if a user is creating a variable of size 2 or 3, it will work in Teiid but not in the VS Code extension.

My whole point was to provide validation that makes sense, that is enforced at Runtime.

Reply:
Yes, I got your point Aurelien. It is just that such a question can be answered at any time later in the process of developing the extension.  For now we defined 4 to 254 chars. You can raise a ticket where we can discuss changing that at a later point. 
Btw. I just check the environment variables in my Linux and the shortest names are 3-4 characters while there were just 2 three letter names in there which were kind of OS related.
( PWD and ZSH ). Most names where between 4 and 24 characters long. I know its not directly comparable but it gives a feeling what might be a real world example.

Reply:
see https://issues.redhat.com/browse/FUSETOOLS2-422

Reply:
Thanks!